test/unit/decorators/hooks.test.ts

```diff
@@ -0,0 +1,80 @@
+import { Model } from '@sequelize/core';
+import type { ModelHooks } from '@sequelize/core/_non-semver-use-at-your-own-risk_/model-typescript.js';
+import {
+  AfterBulkCreate,
+  AfterBulkDestroy,
+  AfterBulkRestore,
+  AfterBulkUpdate,
+  AfterCreate,
+  AfterDestroy,
+  AfterFind,
+  AfterRestore,
+  AfterSave,
+  AfterUpdate,
+  AfterUpsert,
+  AfterValidate,
+  BeforeBulkCreate,
+  BeforeBulkDestroy,
+  BeforeBulkRestore,
+  BeforeBulkUpdate,
+  BeforeCount,
+  BeforeCreate,
+  BeforeDestroy,
+  BeforeFind,
+  BeforeFindAfterExpandIncludeAll,
+  BeforeFindAfterOptions,
+  BeforeRestore,
+  BeforeSave,
+  BeforeUpdate,
+  BeforeUpsert,
+  BeforeValidate,
+} from '@sequelize/core/decorators-legacy';
+import { expect } from 'chai';
+
+// map of hook name to hook decorator
+const hookMap: Partial<Record<keyof ModelHooks, Function>> = {
```

Comment:
`ValidationFailed` and the two `Sync` ones are missing here (apart from the `Associate` ones not even implemented)

Reply:
Done